File: med-app/src/views/UserView.ts
import { UserManager } from '../controllers/UserManager';

export class UserView {
  constructor(private gerenciador: UserManager) {}

  enviarInfoUser(usuario: string, senha: string): string {
    return this.gerenciador.cadastraUsuarioPorCredenciais(usuario, senha);
  }

  mostrarListaUser(): string {
    return this.gerenciador.listarUsuarios();
  }
}

File: med-app-java/src/main/java/com/medapp/views/UserView.java
package com.medapp.views;

import com.medapp.controllers.UserManager;

public class UserView {
    private UserManager gerenciador;

    public UserView(UserManager gerenciador) {
        this.gerenciador = gerenciador;
    }

    public String enviarInfoUser(String usuario, String senha) {
        return gerenciador.cadastraUsuarioPorCredenciais(usuario, senha);
    }

    public String mostrarListaUser() {
        return gerenciador.listarUsuarios();
    }
}
